app/master/cluster_master.py

```diff
@@ -179,7 +179,7 @@ def _disconnect_slave(self, slave):
         """
         # Mark slave dead. We do not remove it from the list of all slaves. We also do not remove it from idle_slaves;
         # that will happen during slave allocation.
-        slave.is_alive = False
+        slave.set_is_alive(False)
```

Comment:
Ok, this is good info. Let me think about this for a bit. I'll hold off on writing tests and merging anything yet.